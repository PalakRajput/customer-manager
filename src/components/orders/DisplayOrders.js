
import React, { useEffect } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { getCustomerAction } from '../../redux/actions/CustomerActions';
import { useParams } from 'react-router-dom';
import DisplayOrdersConditionally from './DisplayOrdersConditionally';



const DisplayOrders = () => {
    const dispatch = useDispatch();
    const customers = useSelector((state) => state.customer.customers);

    const params = useParams();
    useEffect(() => {
        if (customers.length === 0) {
            dispatch(getCustomerAction())
        }
    })
    return (
        // eslint-disable-next-line
        params.id !== undefined ? (customers.length !== 0 ?<DisplayOrdersConditionally customerProp={[customers.find(c => {  
            if (c.id === params.id) {
                return [c]; 
            }

        })]} /> : null)
            : <DisplayOrdersConditionally customerProp={customers} />
    )
}

export default DisplayOrders;