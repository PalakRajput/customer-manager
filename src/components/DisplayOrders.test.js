import { render, screen } from '@testing-library/react'
import DisplayOrders from './DisplayOrders'
import React from 'react'

describe("Display orders ", () => {
    it (" should render correctly ", () => {
           const {asFragment} =  render(<DisplayOrders />)
           expect(asFragment(<DisplayOrders />)).toMatchSnapshot();
    })

    it (" should have h2 with text Orders ", () => {
        render(<DisplayOrders />)
        const heading = screen.getByText(/orders/i)
        expect(heading).toBeInTheDocument();
    })
})