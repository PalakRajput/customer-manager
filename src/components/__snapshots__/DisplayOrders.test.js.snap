// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display orders   should render correctly  1`] = `
<DocumentFragment>
  <h2>
    Orders
  </h2>
</DocumentFragment>
`;
